src/Tokenizer/Transformer/FirstClassCallableTransformer.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tokenizer\Transformer;
+
+use PhpCsFixer\Tokenizer\AbstractTransformer;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @internal
+ */
+final class FirstClassCallableTransformer extends AbstractTransformer
```

Comment:
I will trigger the discussion and sorry that I cannot really discuss it on time.
We should consider if we want to treat this as BC break. if I understand right, it only affects 8.1+, which we don't officially support yet - so maybe not a bc breaker.

Reply:
Yeah, same with non-CT `T_READONLY` token, existing fixers might break, but wasn't considered a BC break by PHP team when aiming for PHP8.1. We could only allow PHP8.1 on next major, I'm +0 for that, although allowing PHP8.1 on current is more easy to keep going with development.

Reply:
Thinking a bit more on it, I think this is not a BC break.

Going with two examples where we change logic when supporting PHP8.1
- Token::isKeyword
- passing Tokens to fixers (including 3rd party) with new token types

The first example; 
on pre PHP8.1 if you pass the value for `T_ENUM` it will return `false`, when running same code on PHP8.1 it will returning `true`. Is this a BC break? I think not, because I expect the method to return if something is a keyword or not based on the PHP version I'm running, so not based on the PHP-CS-Fixer version I'm running. 
Others may think different and expect `false` to be returned for `T_ENUM` for all PHP-CS-Fixer 3.x versions and only `true` on 4.x. However, should 4.x than always return `true` for `T_ENUM`, even for pre PHP8.1 code?

The second example, for parsing PHP8.1 token to the fixers.
Currently, since we already claim to support PHP8.1 in composer (I really don't like it, but here we are), we already pass `T_ENUM` to the fixers. We can see this as a BC break, but it is already to late, as the BC breakage will happen by PHP 8.1 release, not PHP-CS-Fixer release. It can "fixed" by dropping stating we support PHP 8.1 in composer config and ship it next release. If so than this code is not a BC break, as the transformer won't be called. 
However, if we keep claiming PHP8.1 already supported than (as stated) we already pass new tokens like `T_ENUM`, `T_REAONLY` etc. to the fixers, so passing any new custom token types is no different.

I would recommend:
- stop claiming support for any PHP version that is not released yet, for both PHP8.1 case and any other case in the future
- do not consider adding support for new PHP version a BC break and document this
- recommend 3rd party fixers developers to write truthful `isCandidate`  checks and do a runtime PHP version check in it. Also they should set there PHP requirements in composer.json

Reply:
I think this sounds reasonable. 👍